src/util/string.cpp

```diff
@@ -164,6 +174,29 @@ std::string wide_to_utf8(const std::wstring &input)
 	return out;
 }
 
+void wide_add_codepoint(std::wstring &result, char32_t codepoint)
+{
+	if (codepoint < 0x10000) {
+		if (0xD800 <= codepoint && codepoint <= 0xDFFF) {
+			// Invalid codepoint, part of a surrogate pair
+			// Replace with unicode replacement character
+			result.push_back(0xFFFD);
+			return;
+		} 
+		result.push_back((wchar_t) codepoint);
+		return;
+	}
+	codepoint -= 0x10000;
+	if (codepoint >= 0x100000) {
+		// original codepoint was above 0x10FFFF, so invalid
+		// replace with unicode replacement character
+		result.push_back(0xFFFD);
+		return;
+	}
+	result.push_back((wchar_t) ((codepoint >> 10) | 0xD800));
+	result.push_back((wchar_t) ((codepoint & 0x3FF) | 0xDC00));
+}
```

Comment:
The assumption here seems to be that `wchar_t` is 2 bytes only on Windows and 4 bytes on all other platforms.
I don't think this is a good idea.